app/lib/common/providers/room_providers.dart

```diff
@@ -76,6 +78,57 @@ final briefRoomItemWithMembershipProvider =
   );
 });
 
+final briefRoomItemsWithMembershipProvider = FutureProvider.autoDispose<List<RoomItem>>((ref) async {
+  final chatList = ref.watch(chatsProvider).where((element) => (!element.isDm())).toList();
+
+  List<RoomItem> items = [];
+  for (final element in chatList) {
+    final room = await ref.watch(maybeRoomProvider(element.getRoomIdStr()).future);
+    if (room == null) {
+      throw 'Room ${element.getRoomIdStr()} not found';
+    }
+    final profileData = await ref.watch(roomProfileDataProvider(element.getRoomIdStr()).future);
+
+    final item = RoomItem(
+      roomId: room.roomIdStr(),
+      room: room,
+      membership: room.isJoined() ? await room.getMyMembership() : null,
+      activeMembers: [],
+      roomProfileData: profileData,
+    );
+    items.add(item);
+  }
+  return items;
+});
+
+final roomSearchedChatsProvider = FutureProvider.autoDispose<List<RoomItem>>((ref) async {
+  final allRooms = ref.watch(chatsProvider).where((element) => (!element.isDm())).toList();
+  final searchValue = ref.watch(chatSearchValueProvider);
+  final foundRooms = List<RoomItem>.empty(growable: true);
+
+  for (final element in allRooms) {
+    final room = await ref.watch(maybeRoomProvider(element.getRoomIdStr()).future);
+    if (room == null) {
+      throw 'Room ${element.getRoomIdStr()} not found';
+    }
+
+    final profileData = await ref.watch(roomProfileDataProvider(element.getRoomIdStr()).future);
+    final name = profileData.displayName ?? element.getRoomIdStr();
+
+    if (name.toLowerCase().contains(searchValue!.toLowerCase())) {
```

Comment:
```searchValue``` has potential for throwing null, how about handling it prior to reaching this line...

Reply:
Good notice. 👍 
I have will do the required changes here.